synapse/http/matrixfederationclient.py

```diff
@@ -184,20 +184,54 @@ def get_json(self) -> Optional[JsonDict]:
         return self.json
 
 
-class JsonParser(ByteParser[Union[JsonDict, list]]):
+class _BaseJsonParser(ByteParser[T]):
     """A parser that buffers the response and tries to parse it as JSON."""
 
     CONTENT_TYPE = "application/json"
 
-    def __init__(self) -> None:
+    def __init__(self, validator: Optional[Callable[[Any], bool]] = None) -> None:
```

Comment:
Could we have the validator accept `object` rather than `Any`?  It won't make a difference here, but in general I prefer `object`, since `Any` effectively opts out of typechecking.

Reply:
I think so? It would need to be `Optional[object]` though, I think.